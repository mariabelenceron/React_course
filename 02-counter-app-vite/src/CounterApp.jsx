import { useState } from 'react'  //Hook

import PropTypes from 'prop-types'



// Functional Component
export const CounterApp = ({value}) => {

    const [ counter, setCounter ] = useState(value)
    
    const handleAdd = () => {
        // setCounter(counter + 1)
        setCounter((c) => c + 1)
    }
    
    const handleSubtract = () => setCounter(counter - 1)
    
    const handleReset = () => setCounter(value)

    return ( 
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={ handleSubtract }>-1</button>
            <button aria-label='btn-reset' onClick={ handleReset }>Reset</button>
            <button onClick={ handleAdd }> +1 </button>
        </>
    );
}


// Configuraciones
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}