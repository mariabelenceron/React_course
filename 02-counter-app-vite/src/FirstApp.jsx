import PropTypes from 'prop-types';

//--------- Variables
const newMessage = {
    name: 'Belen',
    note: 'Es mi primera app en React'
};

//--------- Functions: Ponerlas afuera para evitar que el FC la renderice de nuevo y utilice mas espacio de memoria
const greeting = ({name, note}) => {
    return `Hola ${name}, tu nota es: ${note}!`
}

// Functional Component
// props : Un objeto, permite establecer una comunicacion entre el componente padre hacia el componente hijo
export const FirstApp = ( {title, subtitle} ) => {
    // console.log(props);

    return (
        // Fragmento
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{ greeting(newMessage) }</p>
        </>
    )
}


// Configuraciones al componente
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subtitle: 'No hay subtitulo'
}