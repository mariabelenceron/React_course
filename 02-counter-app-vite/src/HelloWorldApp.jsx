// Functional Component
export const HelloWorldApp = () => {
    return (
        <div>HelloWorldApp</div>
    )
}
