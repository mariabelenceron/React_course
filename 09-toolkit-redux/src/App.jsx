import { useDispatch, useSelector } from 'react-redux'
import './App.css'
import { decrement, increment, incrementByAmount } from './store/slices/counter'

function App() {

  const { counter } = useSelector(state => state.counter)
  const dispatch = useDispatch()

  return (
    <>
      <h1>Vite + React</h1>
      <div className="card">
        <p>count is {counter}</p>
        <button onClick={() => dispatch(increment())}>
          Increment
        </button>
        <button onClick={() => dispatch(decrement())}>
          Decrement
        </button>
        <button onClick={() => dispatch(incrementByAmount(2))}>
          Increment by 2
        </button>
      </div>
    </>
  )
}

export default App
