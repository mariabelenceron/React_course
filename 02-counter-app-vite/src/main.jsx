import React from 'react';

import ReactDOM from 'react-dom/client'; // Se usa para renderizar la aplicacion

import {CounterApp} from './CounterApp';
import { FirstApp } from './FirstApp';

// import { HelloWorldApp } from './HelloWorldApp';

import './styles.css'; //Tener estilos en toda la aplicaci√≥n


// Renderizar el documento
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* Properties (title) 
            boolean = true de manera implicita
            */}

        {/* <FirstApp title="Hola" subtitle="Soy Belen!"/> */}

        <CounterApp value={1}/>
    </React.StrictMode>
)
