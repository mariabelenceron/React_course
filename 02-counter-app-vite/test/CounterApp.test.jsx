import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas al archivo CounterApp', () => {

    const value = 1;

    test('Debe de hacer match con el snapshot', () => {
        const {container} = render(<CounterApp value={value}/>);
        expect(container).toMatchSnapshot();
    });
    
    test('Debe de mostrar el valor inicial de 100', () => {
        render(<CounterApp value={100}/>);
        expect(screen.getByText(100)).toBeTruthy();
        // expect(Number(screen.getByRole("heading", {level:2}).innerHTML)).toBe(value);
    });

    test('debe de incrementar con el boton +1', () => {
        render(<CounterApp value={value}/>);
        fireEvent.click( screen.getByText('+1') )
        expect(screen.getByText('2')).toBeTruthy();
    });
    
    test('debe de decrementar con el boton -1', () => {
        render(<CounterApp value={value}/>);
        fireEvent.click( screen.getByText('-1') )
        expect(screen.getByText('0')).toBeTruthy();
    });
    
    test('debe de funcionar el boton Reset', () => {
        render(<CounterApp value={value}/>);
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        // fireEvent.click( screen.getByText('Reset') )
        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}))
        expect(screen.getByText(1)).toBeTruthy();
    });
});