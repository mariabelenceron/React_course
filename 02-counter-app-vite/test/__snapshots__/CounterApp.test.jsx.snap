// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al archivo CounterApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    1
     
  </h2>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button>
     +1 
  </button>
</div>
`;
